### ISSUE TITLE
Ringbuffer error on Ubuntu 20.04, kernel version 5.8

### ISSUE THREAD

**Issue by mazahirhussain** [Andreagit97/BPF_ringbuf#1]
The handle_event function is not working for cases (1,3,4) multi_ringbuf_reserve

**Comment by Andreagit97** [comment]
ei @mazahirhussain sorry I didn't see this issue!
I just tried `sudo ./average.sh case3 5 10 no 3` on my local machine
```
Linux 6.2.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC x86_64 x86_64 GNU/Linux
```
and it seems to work well
```
CONFIGURATION: 
------------------------------------------
| 1. Case under test:         -> 'case3'
| 2. Number of stress-tester: ->  5
| 3. Seconds of testing:      ->  10s
| 4. Dedicated cpus:          -> 'no'
| 5. Number of repetition:    ->  3
------------------------------------------

RESULTS: 
*****************************
- Drops: 16462417
- Elaborated: 76843230
*****************************
```

Could you add some more context on why the `handle_event` function doesn't work for you?

**Comment by mazahirhussain** [comment]
If you just try to store the data in "int handle_event(void *ctx, void *data, size_t data_sz)" into a file or just try to print out a message from the function, the whole program gets stuck for all the cases except case2.

**Comment by Andreagit97** [comment]
Uhm considering the examples in `multi_ringbuf_reserve` folder, the `case2` is the only test case that uses the `int handle_event(void *ctx, void *data, size_t data_sz)` method. `case1`/`3`/`4` only initialize this function but it is never called. If you add something like this into `multi_ringbuf_reserve/shared_code.h`
```c
static int handle_event(void *ctx, void *data, size_t data_sz)
{
	printf("sz: %ld\n", data_sz);
	return 0;
}
```

you will see that only `case2` will print something.

BTW dumping on a file or printing something with high rates is something not feasible with these examples, you will experiment huge drops, you could try to use a dedicated thread to print the data but not sure this will solve your issue, probably you should try to reduce the number of event to process with some filter...

**Comment by mazahirhussain** [comment]
Yes, you’re right. I experienced huge drops with case2 when I utilize the handle_event function to dump on to file. Thank you.
